hamilton/function_modifiers/__init__.py

```diff
@@ -88,6 +88,7 @@
 
 subdag = recursive.subdag
 parameterized_subdag = recursive.parameterized_subdag
+with_columns = recursive.with_columns
```

Comment:
This should probably live in a pandas extension -- while pandas has, historically, been the dependency for Hamilton, we want to move it out (and this is very pandas-specific logic)